### ISSUE TITLE
Does not build against boost 1.59

### ISSUE THREAD

**Issue by donSchoe** [ethereum/webthree-umbrella#78]
i can not compile webthree-umbrella agains boost version 1.59 on both the 1.0rc2 tag and the latest github develop snapshot (tag+57). here are the logs https://gist.github.com/donSchoe/38d51f75bdd4e327e3c8

i'm on archlinux kernel 4.2.3-1-ARCH x86_64 GNU/Linux. was working fine with boost 1.58 on both versions.

please check the logs and let me know if you need more details.

**Comment by donSchoe** [comment]
webthree-helpers also fail to compile, too, related?

log https://gist.github.com/donSchoe/4c032db8b558b6dcb269

**Comment by LefterisJP** [comment]
Hey Fellow ArchLinux user!

How are you compiling? Are you following the generic Linux building guide? The log's errors all seem to be boost related. Can you build fine with boost 1.58.X?

I just checked and it seems that boost 1.59.0 is released for ArchLinux today

```
 lefteris  ~  yaourt -Qu
extra/boost 1.58.0-3 -> 1.59.0-2
extra/boost-libs 1.58.0-3 -> 1.59.0-2
extra/evince 3.18.0-1 -> 3.18.1-1
extra/git 2.6.1-1 -> 2.6.2-1
community/gource 0.43-5 -> 0.43-6
extra/libinput 1.0.1-1 -> 1.0.2-1
extra/libtiff 4.0.4-1 -> 4.0.6-1
community/mongodb 3.0.7-1 -> 3.0.7-2
community/ogre 1.9.0-10 -> 1.9.0-11
community/udiskie 1.3.0-2 -> 1.3.2-1
extra/wavpack 4.75.0-1 -> 4.75.2-1
```

If so and if the errors you got are indeed due to the new boost then I will also get them and will work to fix them :)

**Comment by donSchoe** [comment]
Hello! Sorry, I was a bit rushed yesterday because I was at the ethereum meetup in berlin and all of a sudden my complete c++ toolkit for ethereum wasnt working anymore as it was linked against boost 1.58. (errors like "undefinde reference to libboost_thread.so.1.58" etc.)

when i tried to recompile against 1.59 everything broke. logs as above. I used the scripts from the arch user repository. here is the general approach:

release:

```
  git checkout 1.0rc2
  git submodule update --init --recursive

  # Fix miniupnpc issue in submodule
  pushd libweb3core
  git cherry-pick 3ae4d8a
  popd

  mkdir -p build && pushd build
  cmake .. -DCMAKE_INSTALL_PREFIX=/usr \
           -DCMAKE_BUILD_TYPE=Release
  make
  popd
```

develop:

```
  git submodule update --init --recursive
  mkdir -p build && pushd build
  cmake .. -DCMAKE_INSTALL_PREFIX=/usr \
           -DCMAKE_BUILD_TYPE=Release
  make
  popd
```

since it successfully compiled before i can not see any other issues and am pretty sure it's the boost update.

**Comment by donSchoe** [comment]
And btw I'm maintaining both aur-packages in arch, so I'm in trouble now ;-)

https://aur.archlinux.org/packages/ethereum/
https://aur.archlinux.org/packages/ethereum-git/

**Comment by LefterisJP** [comment]
Oh you were in the office yesterday? We should have met.  I am the only core dev using ArchLinux :)

Anyway, it looks like it is indeed boost. I will take a look as soon as I find time, since I am also in the middle of a different task.

**Comment by donSchoe** [comment]
Oh, I should have asked! Maybe we'll meet up for a coffee some day. I'm currently in the process to brush up all the orphaned ethereum packages in the user repository.

Concerning the issue, I would look into the code but the build process is quite complex to dive into it quickly. I noticed some packages (as ethminer for instance) build just fine. So it could be a localized problem. But on the other hand I also noticed it fails on differnt steps using different build processes (e.g. umbrella vs. helper)...

**Comment by donSchoe** [comment]
libethereum can be 'fixed' by commenting out following lines 81-86:
https://github.com/ethereum/libethereum/blob/develop/test/boostTest.cpp#L81-L86

**Comment by donSchoe** [comment]
solidity fails on tests too, commented out lines 49, 53-54, 75 and 79-80 for now:
https://github.com/ethereum/solidity/blob/develop/test/TestHelper.h#L49
https://github.com/ethereum/solidity/blob/develop/test/TestHelper.h#L53-L54
https://github.com/ethereum/solidity/blob/develop/test/TestHelper.h#L75
https://github.com/ethereum/solidity/blob/develop/test/TestHelper.h#L79-L80

**Comment by donSchoe** [comment]
Ok, good news, after removing the lines mentioned above, it compiles.

```
[  2%] Built target scrypt
[  3%] Built target secp256k1
[  3%] Built target dev_BuildInfo.h
[  3%] Built target ethereum_BuildInfo.h
[  3%] Built target cpp-ethereum_BuildInfo.h
[  8%] Built target evmjit
[  9%] Built target ethash
[ 10%] Built target natspec
[ 10%] Built target Sentinel.jsonstub
[ 10%] Built target minerphonehome.jsonstub
[ 10%] Built target minerfarm.jsonstub
[ 10%] Built target webthree_BuildInfo.h
[ 10%] Built target jsonrpcstub
[ 10%] Built target solidity_BuildInfo.h
[ 10%] Built target alethzero_BuildInfo.h
[ 10%] Automatic moc and uic for target aleth
[ 10%] Built target aleth_automoc
[ 10%] Automatic moc, uic and rcc for target alethfive
[ 10%] Automatic moc, uic and rcc for target alethone
[ 10%] Built target alethfive_automoc
[ 10%] Automatic moc and uic for target alethzero
[ 10%] Built target alethone_automoc
[ 10%] Built target alethzero_automoc
[ 10%] Automatic moc and uic for target mix
[ 10%] Built target mix_automoc
[ 15%] Built target devcore
[ 15%] Built target mix_BuildInfo.h
[ 16%] Built target ethash-cl
[ 16%] Built target evmcore
[ 18%] Built target jsengine
[ 21%] Built target devcrypto
[ 24%] Built target evmasm
[ 28%] Built target p2p
[ 28%] Built target rlp
[ 32%] Built target ethcore
[ 33%] Built target lll
[ 35%] Built target whisper
[ 36%] Built target jsconsole
[ 43%] Built target solidity
[ 48%] Built target testweb3core
[ 50%] Built target evm
[ 50%] Built target ethkey
[ 50%] Built target lllc
[ 51%] Built target ethminer
[ 51%] Built target solc
[ 52%] Built target soljson
[ 58%] Built target ethereum
[ 58%] Built target ethvm
[ 60%] Built target testutils
[ 61%] Built target webthree
[ 65%] Built target soltest
[ 66%] Built target web3jsonrpc
[ 66%] Built target exp
[ 72%] Built target testeth
[ 73%] Built target web3testutils
[ 74%] Built target testweb3
[ 78%] Built target aleth
[ 78%] Built target eth
[ 80%] Built target ethrpctest
[ 88%] Built target mix
[ 89%] Built target alethfive
[ 90%] Built target alethone
[100%] Built target alethzero
```

Bad news, you have to find a _real_ fix now :-)

**Comment by LefterisJP** [comment]
@donSchoe Latest develop should now build with boost 1.59

**Comment by donSchoe** [comment]
thanks for the quick fix. just confirmed - working.

**Comment by 5chdn** [comment]
this fix didnt make it into v1.0.1 - any plans to bring this to the release channel?
